pkg/sql/schemachanger/scpb/element_attrs.go

```diff
@@ -0,0 +1,351 @@
+
+// Attributes is part of the eav.Entity interface.
+func (m *RelationDependedOnBy) Attributes() eav.OrdinalSet { return relationDependedOnByAttributes }
+
+// Attributes is part of the eav.Entity interface.
+func (m *SequenceOwnedBy) Attributes() eav.OrdinalSet { return sequenceOwnedByAttributes }
+
+// Attributes is part of the eav.Entity interface.
+func (m *Type) Attributes() eav.OrdinalSet { return typeAttributes }
+
+// Attributes is part of the eav.Entity interface.
+func (m *Schema) Attributes() eav.OrdinalSet { return schemaAttributes }
+
+// Attributes is part of the eav.Entity interface.
+func (m *Database) Attributes() eav.OrdinalSet { return databaseAttributes }
+
+// Get is part of the eav.Entity interface.
+func (m *Column) Get(a eav.Attribute) eav.Value {
+	switch a {
+	case AttrElementType:
+		return getElementType(m)
+	case AttrDescID:
+		return (*eav.Uint32)(&m.TableID)
+	case AttrColumnID:
+		return (*eav.Uint32)(&m.Column.ID)
+	case AttrName:
+		return (*eav.String)(&m.Column.Name)
+	default:
+		return nil
+	}
+}
+
+// Get is part of the eav.Entity interface.
+func (m *PrimaryIndex) Get(a eav.Attribute) eav.Value {
+	switch a {
+	case AttrElementType:
+		return getElementType(m)
+	case AttrDescID:
+		return (*eav.Uint32)(&m.TableID)
+	case AttrIndexID:
+		return (*eav.Uint32)(&m.Index.ID)
+	case AttrName:
+		return (*eav.String)(&m.Index.Name)
+	default:
+		return nil
+	}
+}
+
+// Get is part of the eav.Entity interface.
+func (m *SecondaryIndex) Get(a eav.Attribute) eav.Value {
+	switch a {
+	case AttrElementType:
+		return getElementType(m)
+	case AttrDescID:
+		return (*eav.Uint32)(&m.TableID)
+	case AttrIndexID:
+		return (*eav.Uint32)(&m.Index.ID)
+	case AttrName:
+		return (*eav.String)(&m.Index.Name)
+	default:
+		return nil
+	}
+}
+
+// Get is part of the eav.Entity interface.
+func (m *SequenceDependency) Get(a eav.Attribute) eav.Value {
+	switch a {
+	case AttrElementType:
+		return getElementType(m)
+	case AttrDescID:
+		return (*eav.Uint32)(&m.SequenceID)
+	case AttrReferencedDescID:
+		return (*eav.Uint32)(&m.TableID)
+	case AttrColumnID:
+		return (*eav.Uint32)(&m.ColumnID)
+	default:
+		return nil
+	}
+}
+
+// Get is part of the eav.Entity interface.
+func (m *UniqueConstraint) Get(a eav.Attribute) eav.Value {
+	switch a {
+	case AttrElementType:
+		return getElementType(m)
+	case AttrDescID:
+		return (*eav.Uint32)(&m.TableID)
+	case AttrIndexID:
+		return (*eav.Uint32)(&m.IndexID)
+	default:
+		return nil
+	}
+}
+
+// Get is part of the eav.Entity interface.
+func (m *CheckConstraint) Get(a eav.Attribute) eav.Value {
+	switch a {
+	case AttrElementType:
+		return getElementType(m)
+	case AttrDescID:
+		return (*eav.Uint32)(&m.TableID)
+	case AttrName:
+		return (*eav.String)(&m.Name)
+	default:
+		return nil
+	}
+}
+
+// Get is part of the eav.Entity interface.
+func (m *Sequence) Get(a eav.Attribute) eav.Value {
```

Comment:
Not knowing much about this code, it seems very boiler plate. Should we code generate it instead? I'm thinking about subtle bugs cropping up, or people forgetting to update some of these declarations.